#! /bin/sh /usr/share/dpatch/dpatch-run
## 04_drop_usage_of_fstype.dpatch <mika@grml.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Avoid the use of fstype in a running live system, as it's
## DP: a executable and not a shell function and causes problems
## DP: for example within live-snapshot

@DPATCH@

--- a/scripts/live-helpers
+++ b/scripts/live-helpers
@@ -76,24 +76,6 @@ is_supported_fs ()
 
 get_fstype ()
 {
-	local FSTYPE
-	local FSSIZE
-
-	# fstype misreports LUKS devices
-	if is_luks "${1}"
-	then
-	    /lib/udev/vol_id -t ${1} 2>/dev/null
-	    return
-	fi
-
-	eval $(fstype ${1} 2>/dev/null)
-
-	if [ "${FSTYPE}" != "unknown" ]
-	then
-		echo ${FSTYPE}
-		return 0
-	fi
-
 	/lib/udev/vol_id -t ${1} 2>/dev/null
 }
 
